File: src/shipit/providers/registry.py
from .base import Provider
from .hugo import HugoProvider
from .laravel import LaravelProvider
from .mkdocs import MkdocsProvider
from .node_static import NodeStaticProvider
from .wordpress import WordPressProvider
from .php import PhpProvider
from .python import PythonProvider
from .jekyll import JekyllProvider
from .staticfile import StaticFileProvider


def providers() -> list[type[Provider]]:
    # Order matters: more specific providers first
    return [
        LaravelProvider,
        HugoProvider,
        MkdocsProvider,
        PythonProvider,
        WordPressProvider,
        PhpProvider,
        NodeStaticProvider,
        JekyllProvider,
        StaticFileProvider,
    ]

File: rust/src/provider/registry.rs
//! Provider registry ordering and construction.
//!
//! This will list all provider descriptors once ported.

use crate::provider::ProviderDescriptor;
use crate::provider::hugo::HugoDescriptor;
use crate::provider::jekyll::JekyllDescriptor;
use crate::provider::laravel::LaravelDescriptor;
use crate::provider::mkdocs::MkdocsDescriptor;
use crate::provider::node_static::NodeStaticDescriptor;
use crate::provider::php::PhpDescriptor;
use crate::provider::python::PythonDescriptor;
use crate::provider::staticfile::StaticFileDescriptor;
use crate::provider::wordpress::WordPressDescriptor;

/// Return registered providers in priority order.
pub fn providers() -> Vec<Box<dyn ProviderDescriptor>> {
    vec![
        Box::new(LaravelDescriptor),
        Box::new(HugoDescriptor),
        Box::new(MkdocsDescriptor),
        Box::new(PythonDescriptor),
        Box::new(WordPressDescriptor),
        Box::new(PhpDescriptor),
        Box::new(NodeStaticDescriptor),
        Box::new(JekyllDescriptor),
        Box::new(StaticFileDescriptor),
    ]
}
